pkg/lib/interaction/nodes/create_authenticator_magic_link_otp.go

```diff
@@ -0,0 +1,55 @@
+package nodes
+
+import (
+	"github.com/authgear/authgear-server/pkg/lib/authn"
+	"github.com/authgear/authgear-server/pkg/lib/authn/authenticator"
+	"github.com/authgear/authgear-server/pkg/lib/interaction"
+)
+
+func init() {
+	interaction.RegisterNode(&NodeCreateAuthenticatorMagicLinkOTP{})
+}
+
+type InputCreateAuthenticatorMagicLinkOTP interface {
+	GetMagicLinkOTP() string
+}
+
+type EdgeCreateAuthenticatorMagicLinkOTP struct {
+	Stage         authn.AuthenticationStage
+	Authenticator *authenticator.Info
+}
+
+func (e *EdgeCreateAuthenticatorMagicLinkOTP) Instantiate(ctx *interaction.Context, graph *interaction.Graph, rawInput interface{}) (interaction.Node, error) {
+	var input InputCreateAuthenticatorMagicLinkOTP
+	if !interaction.Input(rawInput, &input) {
+		return nil, interaction.ErrIncompatibleInput
+	}
+
+	// Unlike Whatsapp, magic link is already verified before this edge, so we directly go to next node
```

Comment:
We cannot skip the verification here, and let the user simply submit the `matched` action and go to the next step. There will be a vulnerability that users don't need the code and can bypass the checking easily.

We can check the code in the email, but it needs to be dry run. And we must check the code again in the original web session.

In whatsapp otp implementation, we store the user inputted code from the webhook and trigger web session refresh. In the original web session, it will compare the user inputted code with the real code to ensure a correct code is received. When auto-refresh happens it is dry run only, and the code is consumed when the user clicks `Next`.

Please find me to further explain if it is not clear.